api/_dexes/cross-swap.ts

```diff
@@ -139,4 +154,92 @@ export function getCrossSwapType(params: {
   return CROSS_SWAP_TYPE.ANY_TO_ANY;
 }
 
-export function calcFees() {}
+export async function buildCrossSwapTx(
+  crossSwapQuotes: CrossSwapQuotes,
+  integratorId?: string
+) {
+  const originChainId = crossSwapQuotes.crossSwap.inputToken.chainId;
+  const destinationChainId = crossSwapQuotes.crossSwap.outputToken.chainId;
+  const spokePool = getSpokePool(originChainId);
+  const deposit = {
+    depositor: crossSwapQuotes.crossSwap.recipient,
+    recipient: crossSwapQuotes.destinationSwapQuote
+      ? getMultiCallHandlerAddress(destinationChainId)
+      : crossSwapQuotes.crossSwap.recipient,
+    inputToken: crossSwapQuotes.bridgeQuote.inputToken.address,
+    outputToken: crossSwapQuotes.bridgeQuote.outputToken.address,
+    inputAmount: crossSwapQuotes.bridgeQuote.inputAmount,
+    outputAmount: crossSwapQuotes.bridgeQuote.outputAmount,
+    destinationChainid: crossSwapQuotes.bridgeQuote.outputToken.chainId,
+    exclusiveRelayer:
+      crossSwapQuotes.bridgeQuote.suggestedFees.exclusiveRelayer,
+    quoteTimestamp: crossSwapQuotes.bridgeQuote.suggestedFees.timestamp,
+    fillDeadline: await getFillDeadline(spokePool),
+    exclusivityDeadline:
+      crossSwapQuotes.bridgeQuote.suggestedFees.exclusivityDeadline,
+    message: crossSwapQuotes.bridgeQuote?.message || "0x",
```

Comment:
Per the lines above it seems like we expect `crossSwapQuotes.bridgeQuote` to always be defined.

```suggestion
    message: crossSwapQuotes.bridgeQuote.message || "0x",
```

Reply:
Good point

Reply:
https://github.com/across-protocol/frontend/pull/1274/commits/484314e987aa4b4c8667e0ac06221b30e839ae3f